nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,50 @@
+const { request, response } = require("express");
+const express = require("express");
+const router = express.Router();
+const meals = require("./../data/meals.json");
+
+//Respond with the json for all the meals
+
+router.get("/", async(request, response) => {
+  let data = meals;
+  
+  if ("maxPrice" in request.query) {
+    const maxPrice = Number(request.query.maxPrice);
+    data = data.filter(meal => meal.price <= maxPrice);
+  }
+  
+  if ("title" in request.query) {
+    const text = request.query.title;
+    data = data.filter(meal=> meal.title.toLocaleLowerCase().includes(text))
+  }
+  
+  if ("createdAfter" in request.query) {
+    const d1 = new Date(request.query.createdAfter);
+    data = data.filter(meal=> new Date(meal.createdAt) > d1)
+  }
+  
```

Comment:
Would be good to add a check and an error response if `date` is not a date / if the `new Date ()` fails 💡